osutil/mountentry_linux_test.go

```diff
@@ -512,3 +512,20 @@ func (s *entrySuite) TestXSnapdMustExistDir(c *C) {
 	e = &osutil.MountEntry{Options: []string{osutil.XSnapdMustExistDir("$HOME")}}
 	c.Assert(e.XSnapdMustExistDir(), Equals, "$HOME")
 }
+
+func (s *entrySuite) TestSplitSystemdMountOptions(c *C) {
+	tt := []struct {
+		in  string
+		out []string
+	}{
+		{"foo,bar", []string{"foo", "bar"}},
+		{`foo\,,bar`, []string{`foo,`, "bar"}},
+		{`foo\f,bar`, []string{`foo\f`, "bar"}},
+		{`lowerdir=a\ ,upperdir=b\ ,workdir=c\ `, []string{`lowerdir=a\ `, `upperdir=b\ `, `workdir=c\ `}},
```

Comment:
I had a look at how this works in practice. Is started with directories with a trailing space.

```
zyga@pi5-1:~$ cd overlayfs-experiment/
zyga@pi5-1:~/overlayfs-experiment$ mkdir "a " "b " "c "
zyga@pi5-1:~/overlayfs-experiment$ mkdir a, b, c,
zyga@pi5-1:~/overlayfs-experiment$ sudo mount -t overlay overlayfs /mnt -o 'lowerdir=a ,upperdir=b ,workdir=c '
zyga@pi5-1:~/overlayfs-experiment$ findmnt /mnt
TARGET SOURCE    FSTYPE  OPTIONS
/mnt   overlayfs overlay rw,relatime,lowerdir=a ,upperdir=b ,workdir=c ,uuid=on
zyga@pi5-1:~/overlayfs-experiment$ cat /proc/self/mountinfo | grep /mnt
214 26 0:48 / /mnt rw,relatime shared:275 - overlay overlayfs rw,lowerdir=a\040,upperdir=b\040,workdir=c\040,uuid=on
zyga@pi5-1:~/overlayfs-experiment$ systemctl show mnt.mount | grep ^Options=
Options=rw,relatime,lowerdir=a ,upperdir=b ,workdir=c ,uuid=on
```

This works fine when done by hand (notice how different the escape logic is in each case) but fails with done with a systemd unit:

```
zyga@pi5-1:/tmp$ systemctl cat mnt.mount
# /etc/systemd/system/mnt.mount
[Mount]
What=overlay
Where=/mnt
Type=overlay
Options=lowerdir=/tmp/a\ ,upperdir=/tmp/b\ ,workdir=/tmp/c\ ,rw
```

While this one does not. Notice that there's a space at the end of the file, just before the final newline.

```
zyga@pi5-1:/tmp$ systemctl cat mnt.mount
# /etc/systemd/system/mnt.mount
[Mount]
What=overlay
Where=/mnt
Type=overlay
Options=lowerdir=/tmp/a\ ,upperdir=/tmp/b\ ,workdir=/tmp/c\ 
```